1603 speech to text hotkey

Rrojaski opened the pull request:
### Pull Request Type

<!-- For change type, change [ ] to [x]. -->

- [x] ✨ feat
- [ ] 🐛 fix
- [ ] ♻️ refactor
- [ ] 💄 style
- [ ] 🔨 chore
- [ ] 📝 docs

### Relevant Issues

<!-- Use "resolves #xxx" to auto resolve on merge. Otherwise, please use "connect #xxx" -->

resolves #1603


### What is in this change?

<!-- Describe the changes in this PR that are impactful to the repo. -->

Implemented Ctrl (Cmd on Mac) + 'm' hotkeys to activate the existing React Speech Recognition feature.


### Additional Information

<!-- Add any other context about the Pull Request here that was not captured above. -->

### Developer Validations

<!-- All of the applicable items should be checked. -->

- [x] I ran `yarn lint` from the root of the repo & committed changes
- [x] Relevant documentation has been updated
- [x] I have tested my code functionality
- [x] Docker build succeeds locally

timothycarambat commented:
Love the idea. I have some feedback currently from a UI perspective. When I press CMD+Enter the mic turns on but is not super obvious my hotkey did that (the icon just gets white for the mic). Then I can speak and I see the words coming in, but then I don't know how to really "stop" the streaming other than shutting up for the silence interval.

If I click on the mic icon, it'll auto-send, but if I click the send icon itself it will double-send my query as the mic gets the event to stop listening.

Thinking about a UX perspective that makes this less confusing for someone who might find themselves toying with this hot-key combination

Rrojaski commented:
Thanks for the feedback! It looks like the query was being double-sent by clicking the mic button or by pressing the hotkeys because of a timeout inside the SpeechToText component. I've added an event listener for PROMPT_INPUT_EVENT to check if the user clicks send before the timeout finishes.

I've changed the hotkeys to CMD+SPACE to avoid interfering with what the ENTER key currently does.

I'm still considering a better user experience for stopping the speech recognition. Do you have any recommendations?

Side note: I'm not sure if it's just me, but "SpeechRecognition.stopListening()" doesn't appear to stop the speech recognition.
